tests/lib/devicetree/devices/src/main.c

```diff
@@ -74,23 +74,51 @@ static bool check_handle(device_handle_t hdl,
 	return false;
 }
 
+struct requires_context {
+	uint8_t ndevs;
+	const struct device *rdevs[3];
```

Comment:
Why 3, here and below; isn't 2 enough? I know 3 isn't an error, but I get weirded out by oversized buffers like this; makes me feel like I'm missing something. Am I?

Reply:
I had an idea about checking that it didn't write more than was necessary, but didn't follow through.  On reconsideration that doesn't seem worth doing.